extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -164,112 +187,121 @@ private void restartJobs() {
         if (schedulerFuture != null && !schedulerFuture.isCancelled()) {
             schedulerFuture.cancel(true);
         }
-
-        schedulerFuture = scheduler.schedule(new Runnable() {
-            @Override
-            public void run() {
+        if (localExecutor != null) {
+            localExecutor.shutdownNow();
+        }
+        localExecutor = Executors.newSingleThreadExecutor();
```

Comment:
My bad. I meant 2 seconds. In this handler, `scheduler` primarily schedules a runnable instance that schedules multiple jobs. Inside this runnable, two jobs are executed immediately. I can still invoke these immediate jobs inside the same runnable instance that gets scheduled by the `scheduler`.

```java
scheduler.schedule(() -> {
	.....
	.....
	scheduler.submit(..);
	.....
	scheduler.submit(..);
	.....
});
```

For a cleaner perspective, a single thread executor has been introduced. There is no other reason behind it.